vnext/ReactUWP/Utils/LocalBundleReader.cpp

```diff
@@ -22,9 +23,23 @@ std::future<std::string> LocalBundleReader::LoadBundleAsync(const std::string& b
   co_await winrt::resume_background();
 
   auto file = co_await winrt::Windows::Storage::StorageFile::GetFileFromApplicationUriAsync(uri);
-  auto hdata = co_await winrt::Windows::Storage::FileIO::ReadTextAsync(file);
 
-  co_return facebook::react::unicode::utf16ToUtf8(std::wstring(hdata.data()));
+  // Read the buffer manually to avoid a Utf8 -> Utf16 -> Utf8 encoding roundtrip.
+  auto fileBuffer { co_await winrt::Windows::Storage::FileIO::ReadBufferAsync(file) };
+  auto dataReader{ winrt::Windows::Storage::Streams::DataReader::FromBuffer(fileBuffer) };
+  std::vector<uint8_t>* data = new std::vector<uint8_t>(fileBuffer.Length() + 1);
```

Comment:
when does this get freed? don't see why this is operator new'd, if it needs to be it should be deleted

Reply:
I must have a misunderstanding about the lifetime of vectors (my c++ is rusty 100% ready to admit that).

I thought that without newing the vector would go out of scope and be freed when this function returns. 

Let me give it a shot without the new.

Reply:
std::string is going to make a copy of whatever you give it in its constructor, so the data will still be reachable.  If you wan to avoid that copy you need to pre-reserve space.  We could fix such conversions in a handful of places - this is a very big string so maybe more useful here - but networking, websocket, devsupport could all be improved... the place we do avoid an extra string is in NativeUIManager.cpp YogaLog(), that is following what the STL does when you convert a double to a string.  We can do the same by creating a string with a big enough buffer from creation, copying into &str[0] then resizing so the string is told what its content size is.

Reply:
actually not sure why the 'resize' line is needed.  this is the STL implementation I was referring to:
template <class _Ty>
inline string _Floating_to_string(const char* _Fmt, _Ty _Val) { // convert _Val to string
    static_assert(is_floating_point_v<_Ty>, "_Ty must be floating point");

    const auto _Len = static_cast<size_t>(_CSTD _scprintf(_Fmt, _Val));
    string _Str(_Len, '\0');
    _CSTD sprintf_s(&_Str[0], _Len + 1, _Fmt, _Val);
    return _Str;
}

Reply:
Ah! Thank you for the explanation, very informative!
I will definitely keep the copy reduction in mind. 
For now I am happy with just avoiding the encoding round trip.

Reply:
Updated to pre-allocate the string and avoid the copy. This shaves off another handful of ms from the previous versions.

Maybe you already stated this and I misunderstood. But do you know if there is any mechanism we can use to skip the initial fill of the string?
In the trace you can see the initial memset and moving the file into the string.
![image](https://user-images.githubusercontent.com/48095233/59527423-660be080-8e90-11e9-86ab-6ff20526c076.png)